### PR TITLE
[Enhancement] Handle timeouts in compaction

### PR DESCRIPTION
**Pull request by sduzh**
Implemented timeout processing in compaction_scheduler by setting a default timeout and adding a timeout check function. This helps to improve robustness of compaction in cases of unexpected long execution times.

Also made constants for timeout values in LakeService.java for more clear reading and maintainability.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/lake/compaction_scheduler.h [hunk 1]
```diff
@@ -63,18 +63,30 @@ class CompactionTaskCallback {
...
         std::lock_guard l(_mtx);
         _status.update(st);
     }
 
+    bool timeout_exceeded() const { return butil::gettimeofday_ms() >= _timeout_duetime; }
+
+    int64_t timeout() const;
+
 private:
+    const static int64_t kDefaultTimeout = 24L * 60 * 60 * 1000; // 1 day
```
Comment (imay) [inline be/src/storage/lake/compaction_scheduler.h:81]: Is it necessary to make the default timeout 1 day?

Comment (imay) [reply be/src/storage/lake/compaction_scheduler.h:81]:
```suggestion
    const static int64_t kDefaultTimeoutMs = 24L * 60 * 60 * 1000; // 1 day
```

File: be/src/storage/lake/compaction_scheduler.h [hunk 2]
```diff
@@ -63,18 +63,30 @@ class CompactionTaskCallback {
...
+
     CompactionScheduler* _scheduler;
     mutable StackTraceMutex<bthread::Mutex> _mtx;
     const lake::CompactRequest* _request;
     lake::CompactResponse* _response;
     ::google::protobuf::Closure* _done;
     Status _status;
+    int64_t _timeout_duetime;
```
Comment (imay) [inline be/src/storage/lake/compaction_scheduler.h:89]:
```suggestion
    int64_t _timeout_duetime_ms;
```

File: be/src/storage/lake/compaction_scheduler.h [hunk 3]
```diff
@@ -63,18 +63,30 @@ class CompactionTaskCallback {
...
+    }
+
     void update_status(const Status& st) {
         std::lock_guard l(_mtx);
         _status.update(st);
     }
 
+    bool timeout_exceeded() const { return butil::gettimeofday_ms() >= _timeout_duetime; }
+
+    int64_t timeout() const;
```
Comment (imay) [inline be/src/storage/lake/compaction_scheduler.h:78]:
```suggestion
    int64_t timeout_ms() const;
```

**Review by imay — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**